subsys/bluetooth/controller/hci_internal.c

```diff
@@ -55,51 +61,100 @@ static bool command_generates_command_complete_event(uint16_t hci_opcode)
 	}
 }
 
+static void encode_command_status(uint8_t * const event,
+				  uint16_t hci_opcode,
+				  uint8_t status_code)
+{
+
+	struct bt_hci_evt_hdr *evt_hdr = (struct bt_hci_evt_hdr *)event;
+	struct bt_hci_evt_cmd_status *evt_data =
+			(struct bt_hci_evt_cmd_status *)&event[BT_HCI_EVT_HDR_SIZE];
+
+	evt_hdr->evt = BT_HCI_EVT_CMD_STATUS;
+	evt_hdr->len = sizeof(struct bt_hci_evt_cmd_status);
+
+	evt_data->status = status_code;
+	evt_data->ncmd = 1;
+	evt_data->opcode = hci_opcode;
+}
+
+static void encode_command_complete_header(uint8_t * const event,
+					   uint16_t hci_opcode,
+					   uint8_t param_length,
+					   uint8_t status)
+{
+	struct bt_hci_evt_hdr *evt_hdr = (struct bt_hci_evt_hdr *)event;
+	struct bt_hci_evt_cmd_complete *evt_data =
+			(struct bt_hci_evt_cmd_complete *)&event[BT_HCI_EVT_HDR_SIZE];
+
+	evt_hdr->evt = BT_HCI_EVT_CMD_COMPLETE;
+	evt_hdr->len = param_length;
+	evt_data->ncmd = 1;
+	evt_data->opcode = hci_opcode;
+	event[BT_HCI_EVT_HDR_SIZE + sizeof(struct bt_hci_evt_cmd_complete)] = status;
+}
+
 /* Return true if the host has been using both legacy and extended HCI commands
  * since last HCI reset.
  */
-static bool is_host_using_legacy_and_extended_commands(uint16_t hci_opcode)
+static bool check_and_handle_is_host_using_legacy_and_extended_commands(uint8_t *cmd_in,
+									uint16_t opcode)
 {
 #if defined(CONFIG_BT_HCI_HOST)
 	/* For a combined host and controller build, we know that the zephyr
 	 * host is used. For this case we know that the host is not
 	 * combining legacy and extended commands. Therefore we can
-	 * simplify this validation. */
+	 * simplify this validation.
+	 */
 	return false;
 #else
 	/* A host is not allowed to use both legacy and extended HCI commands.
 	 * See Core v5.1, Vol2, Part E, 3.1.1 Legacy and extended advertising
 	 */
-	static enum {
-		ADV_COMMAND_TYPE_NONE,
-		ADV_COMMAND_TYPE_LEGACY,
-		ADV_COMMAND_TYPE_EXTENDED,
-	} type_of_adv_cmd_used_since_reset;
+	static enum type_of_adv_cmd type_of_adv_cmd_used_since_reset;
```

Comment:
You could consider explicitly initializing `type_of_adv_cmd_used_since_reset` to `ADV_COMMAND_TYPE_NONE`